src/AutoRest.CSharp/Mgmt/Output/MgmtObjectType.cs

```diff
@@ -49,8 +49,14 @@ private HashSet<string> GetParentPropertyNames()
             return EnumerateHierarchy()
                 .Skip(1)
                 .SelectMany(type => type.Properties)
-                .Select(p => p.Declaration.Name)
+                .Select(p => isResourceType(p) ? "Type" : p.Declaration.Name)
                 .ToHashSet();
+
+            // In common-types, "type" is a property to represent resource type
+            // In ResourceManager.ResourceData, this property is named as "ResourceType"
+            // Return "Type" as property name for it to skip it in the output properties
+            bool isResourceType(ObjectTypeProperty property)
```

Comment:
Seems more appropriate to call it `RenameResourceType` and put all the rename logic into it?